src/osd/PG.h

```diff
@@ -197,6 +197,9 @@ class PG {
   void update_snap_mapper_bits(uint32_t bits) {
     snap_mapper.update_bits(bits);
   }
```

Comment:
It's an annoying weakness of C++ that it's hard to indicate with the type system that you are giving ownership of the pointer to the caller.  Let's make it clear by adding

/// get_is_recoverable_predicate: caller owns returned pointer and must delete when done